sdks/python/apache_beam/io/iobase.py

```diff
@@ -176,7 +176,7 @@ def read(self, range_tracker):
 
   def default_output_coder(self):
     """Coder that should be used for the records returned by the source."""
-    return PickleCoder()
+    return coders.registry.get_coder(object)
```

Comment:
Does the default object coder have any restrictions when it comes to types of objects it can support. Currently, by default, we only support picklable objects. Can we give more details here so that users can know what type of objects a source can produce if the source use the default coder ?